common/src/main/java/com/wynntils/models/skillpoint/SavableSkillPointSet.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.skillpoint;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class SavableSkillPointSet {
+    private final int strength;
+    private final int dexterity;
+    private final int intelligence;
+    private final int defence;
+    private final int agility;
+    private final List<String> armourNames;
+    private final List<String> accessoryNames;
+
+    /**
+     * Constructs a new SavableSkillPointSet representing just a loadout.
+     */
+    public SavableSkillPointSet(int[] skillPoints) {
+        this.strength = skillPoints[0];
+        this.dexterity = skillPoints[1];
+        this.intelligence = skillPoints[2];
+        this.defence = skillPoints[3];
+        this.agility = skillPoints[4];
+        this.armourNames = new ArrayList<>();
+        this.accessoryNames = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a new SavableSkillPointSet representing a full build with gear.
+     */
+    public SavableSkillPointSet(int[] skillPoints, List<String> armourNames, List<String> accessoryNames) {
+        this.strength = skillPoints[0];
+        this.dexterity = skillPoints[1];
+        this.intelligence = skillPoints[2];
+        this.defence = skillPoints[3];
+        this.agility = skillPoints[4];
+        this.armourNames = new ArrayList<>(armourNames);
+        this.accessoryNames = new ArrayList<>(accessoryNames);
```

Comment:
These should be immutable lists to prevent any weird cases. That also means you can cache `isBuild`. Whenever you convert a saved skill set to a build, just recreate this object.